lib/sfloat/sfloat.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <inttypes.h>
+#include <stdio.h>
+
+#include <sfloat.h>
+
+#include <zephyr/toolchain.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util_macro.h>
+
+/* Maximum SFLOAT exponent multiplier: 10^7 */
+#define SFLOAT_EXP_MULTIPLIER_MAX (10000000)
+/* Normalized SFLOAT exponent multiplier: 10^-8 */
+#define SFLOAT_EXP_MULTIPLIER_NORMALIZER (100000000)
+
+/* Absolute minimum float resolution. */
+#define FLOAT_ABS_MIN (0.000001f)
+/* Minimum negative float resolution. */
+#define FLOAT_NEG_MIN (SFLOAT_EXP_MULTIPLIER_MAX * (-1.0f) * (1 << 11))
+/* Maximum positive float resolution. */
+#define FLOAT_POS_MAX (SFLOAT_EXP_MULTIPLIER_MAX * (1.0f) * ((1 << 11) - 1))
+
+/* Float type should use binary32 notation from the IEEE 754-2008 specification. */
+BUILD_ASSERT(sizeof(float) == sizeof(uint32_t));
+
+struct sfloat_desc {
+	uint8_t exponent;
+	uint16_t mantissa;
+};
+
+static struct sfloat_desc sfloat_desc_from_float(float float_num)
```

Comment:
Wouldn't be better to return sfloat structure through function parameters and return error code?

Reply:
There is no error code since it is part of the type conversion. Typically, it is a bad idea to return structure through the return value but this type is minimal and its like returning uint32_t.

I would leave it as it now